django/thunderstore/core/management/commands/create_test_data.py

```diff
@@ -35,6 +37,70 @@ def clear(self) -> None:
         ...
 
 
+class CommunityPopulator(ContentPopulator):
+    communities: Optional[List[Community]] = None
+    identifier_prefix = "test-community-"
+    name_prefix = "Test Community "
+
+    def populate(self, context: ContentPopulatorContext) -> None:
+        print("Populating communities...")
+
+        last = 0
+        last_community = Community.objects.order_by("-pk").first()
+        if last_community:
+            last = last_community.pk
+
+        existing = list(
+            Community.objects.filter(name__startswith=self.name_prefix)[
+                : context.community_count
+            ]
+        )
+        remainder = context.community_count - len(existing)
+
+        self.communities = existing + [
+            Community.objects.create(
+                name=f"{self.name_prefix}{last + i}",
+                identifier=f"{self.identifier_prefix}{last + i}",
+            )
+            for i in print_progress(range(remainder), remainder)
+        ]
+
+    def update_context(self, context) -> None:
+        if self.communities is not None:
+            context.communities = self.communities
+        else:
+            context.communities = Community.objects.filter(
+                name__startswith=self.name_prefix
+            )[: context.community_count]
+
+    def clear(self) -> None:
+        print("Deleting existing test communities...")
+        Community.objects.filter(name__startswith=self.name_prefix).delete()
+
+
+class CommunitySitePopulator(ContentPopulator):
+    def populate(self, context: ContentPopulatorContext) -> None:
+        print("Populating community sites...")
+
+        for community in context.communities:
+            community_site = CommunitySite.objects.filter(community=community).first()
+            if community_site:
+                continue
+            CommunitySite.objects.create(
+                community=community,
+                site=Site.objects.create(
+                    name="Thunderstore",
+                    domain=f"{community.identifier}.thunderstore.localhost",
```

Comment:
I was thinking it could perhaps be added as a parameter to the script, but I'll leave that for later if/when it becomes useful. The `thunderstore.localhost` domain is what the readme currently instructs to set up, and the script is meant for local use only at the moment.